pkg/services/publicdashboards/service/service.go

```diff
@@ -342,6 +342,32 @@ func (pd *PublicDashboardServiceImpl) Delete(ctx context.Context, orgId int64, u
 	return nil
 }
 
+func (pd *PublicDashboardServiceImpl) HandleDashboardDeleted(ctx context.Context, dashboard *dashboards.Dashboard) error {
+	if dashboard.IsFolder {
+		// get all pubdashes for the folder
+		pubdashes, err := pd.store.GetByDashboardFolder(ctx, dashboard)
+		if err != nil {
+			return err
+		}
+		// delete each pubdash
+		for _, pubdash := range pubdashes {
+			err = pd.serviceWrapper.HandlePublicDashboardDeleted(ctx, pubdash)
+			if err != nil {
+				return err
+			}
+		}
+
+		return nil
+	}
+
+	pubdash, err := pd.store.FindByDashboardUid(ctx, dashboard.OrgID, dashboard.UID)
+	if err != nil {
+		return err
+	}
+
```

Comment:
If i'm not wrong, this is the code that has to be executed in both scenarios (oss or enterprise). Therefore, it should be in the service wrapper because, if it's enterprise, another service instance will be injected in the api

Reply:
When we delete a dashboard, we always delete the pubdashes for it regardless if its OSS or enterprise. We only handle things differently when we delete a pubdash. So this method doesn't need to change. The different implementations for handling a pubdash deletion are implemented in each wrapper.

Reply:
The [public dashboard service](pkg/services/publicdashboards/service/service.go) has a service wrapper attribute which, depending on if it's oss or enterprise, will be injected with [oss service wrapper](pkg/services/publicdashboards/service/service_wapper.go) or [enterprise service wrapper](src/pkg/extensions/enterprisepublicdashboards/service/service_wrapper.go). The enterprise service_wrapper has an instance of the oss service_wrapper.
So, if it's oss, the public dashboard service will be injected with oss service_wrapper which should have this code.
Otherwise, if it's enterprise, the public dashboard service will be injected with enterprise service_wrapper, which will do what needs to do and then, call the oss service_wrapper.

You can check an example with `FindByDashboardUid` method

Reply:
yes, you're right. sounds good 💪🏽